client/components/section-nav/tabs.jsx

```diff
@@ -42,18 +42,18 @@ class NavTabs extends Component {
 	};
 
 	componentDidMount() {
-		this.setDropdown();
-		this.debouncedAfterResize = debounce( this.setDropdown, 300 );
-
-		window.addEventListener( 'resize', this.debouncedAfterResize );
+		this.setDropdownAfterLayoutFlush();
+		window.addEventListener( 'resize', this.setDropdownDebounced );
+		this._isMounted = true;
 	}
 
 	componentDidUpdate() {
-		this.setDropdown();
+		this.setDropdownAfterLayoutFlush();
 	}
 
 	componentWillUnmount() {
-		window.removeEventListener( 'resize', this.debouncedAfterResize );
+		this._isMounted = false;
```

Comment:
Instead of keeping a flag `_isMounted`, could we cancel the `setTimeout` from `setDropdownAfterLayoutFlush` in `componentWillUnmount`? I think I read somewhere that relying on isMounted flags is an anti-pattern for React.

Reply:
Probably this one? https://reactjs.org/blog/2015/12/16/ismounted-antipattern.html

Reply:
Overall, I think using a flag here is okay, since we're using both an RAF and setTimeout here. It'd be annoying to try and cancel both on unmount.

Reply:
I separated this functionality into a standalone `afterLayoutFlush` module. The created wrapper has a `cancel` method that can be called to cancel the scheduled invocations. Lodash functions like `throttle` or `debounce` have the same API.